src/message.ts

```diff
@@ -414,6 +414,93 @@ export class Message implements Sayable {
     return fromId === userId
   }
 
+  /**
+   *
+   * Get the mentioned contact which the message is mentioned for.
+   * message event table
+   *
+   * |                                                                            | Web  |  Mac PC Client | iOS Mobile |  android Mobile |
+   * | :---                                                                       | :--: |     :----:     |   :---:    |     :---:       |
+   * | [You were mentioned] tip ([有人@我]的提示)                                   |  ✘   |        √       |     √      |       √         |
+   * | Identify magic code (8197) by copy & paste in mobile                       |  ✘   |        √       |     √      |       ✘         |
+   * | Identify magic code (8197) by programming                                  |  ✘   |        ✘       |     ✘      |       ✘         |
+   * | Identify two contacts with the same roomAlias by [You were  mentioned] tip |  ✘   |        ✘       |     √      |       √         |
+   *
+   * @returns {Contact[]} return the contactList which the message is mentioned for
+   *
+   * @example
+   * ```ts
+   * const content = message.content()
+   * const contactList = message.mentioned()
+   * console.log(content)
+   * console.log(contactList)
+   * ```
+   */
+  public mentioned(): Contact[] {
+    let contactList: Contact[] = []
+    const room = this.room()
+    if (this.type() !== MsgType.TEXT || !room ) {
+      return contactList
+    }
+
+    // Example: `Let me introduce @lijiarui@wechaty@beijing , she is a wechaty contributor`
+    // Trying to split array with magic code `8197` and get atStrings
+    // Example result: atStrings = ['Let me introduce @lijiarui@wechaty@beijing', ', she is a wechaty contributor']
+    const atStrings = this.content().split(String.fromCharCode(8197))
+
+    if (atStrings.length === 0) return contactList
+
+    // Using `filter(e => e.indexOf('@') > -1)` to filter the string without `@`
+    // Example result: atStrings.filter(e => e.indexOf('@') > -1) = ['Let me introduce @lijiarui@wechaty@beijing']
+    const strings = atStrings.filter(e => e.indexOf('@') > -1).map(e => {
+
+      // Example result: list = ['Let me introduce ','@lijiarui', '@wechaty', '@beijing']
+      const list = e.split('@')
+
+      // `list.length < 3` means the array just have less than 2 element, means contact doesn't have multiple `@` signs
+      if (list.length < 3) {
+        return [e.slice(e.lastIndexOf('@') + 1)]
+      } else {
+
+        // Deal with contact with multiple `@`signs
+        // Select element from `list` array to join strings with the following rules:
+        // 1. end with the last element from `list` array.
+        // 2. element are continuous, eg: [3],[2,3],[1,2,3]
+
+        // Here: list = ['Let me introduce ','@lijiarui', '@wechaty', '@beijing']
+        let nickList: string[] = []
```

Comment:
```js
const str = '@a@b@c'

let tryName = ''
const tryNameList = []

str.split('@')
  .filter(c => !!c)
  .forEach(c => {
    tryName += '@' + c
    tryNameList.push(tryName.slice(1)) // get rid of the `@` at beginning
  })

console.log(tryNameList)
```

will output: `[ 'a', 'a@b', 'a@b@c' ]`

Please use the above code to replace your for loop.